pkg/controllers/state/statenode.go

```diff
@@ -70,6 +71,13 @@ func (n StateNodes) Pods(ctx context.Context, c client.Client) ([]*v1.Pod, error
 	return pods, nil
 }
 
+// Disruptable filters StateNodes that are meet the IsDisruptable condition
+func (n StateNodes) Disruptable(ctx context.Context, kubeClient client.Client, recorder events.Recorder) StateNodes {
```

Comment:
I don't think that you even need the recorder here anymore given that we are now returning the event back from the caller